sdk/src/utils/io_utils.rs

```diff
@@ -234,16 +236,64 @@ pub fn wasm_remove_dir_all<P: AsRef<std::path::Path>>(path: P) -> Result<()> {
     ))?
 }
 
+/// Convert a URI to a file path using PathBuf for better path handling.
+#[cfg(feature = "file_io")]
+pub fn uri_to_path(uri: &str, manifest_label: Option<&str>) -> PathBuf {
+    let mut path_str = uri.replace(':', "_");
+    if let Some(stripped) = path_str.strip_prefix("self#jumbf=") {
+        path_str = stripped.to_owned();
+    } else {
+        return PathBuf::from(path_str);
+    }
+
+    let mut path = PathBuf::from(path_str);
+
+    if let Ok(stripped) = path.strip_prefix("/c2pa/") {
+        path = stripped.to_path_buf();
+    } else if let Some(manifest_label) = manifest_label {
+        let mut new_path = PathBuf::from(manifest_label);
```

Comment:
The original logic replaced colons and underscores at the end, so that the manifest label was also converted. This logic will not convert the manifest label.